File: Website/backend/ecommerce/src/main/java/com/se/ecommerce/dto/Purchase.java
package com.se.ecommerce.dto;

import com.se.ecommerce.entity.Address;
import com.se.ecommerce.entity.Customer;
import com.se.ecommerce.entity.Order;
import com.se.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: Website/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    order: Order;
    customer: Customer;
    shippingAddress: Address;
    billingAddress: Address;
    orderItems: OrderItem[];
}
